grafanalib/core.py

```diff
@@ -172,14 +172,20 @@ class Legend(object):
     min = attr.ib(default=False, validator=instance_of(bool))
     show = attr.ib(default=True, validator=instance_of(bool))
     total = attr.ib(default=False, validator=instance_of(bool))
-    values = attr.ib(default=False, validator=instance_of(bool))
+    values = attr.ib(default=None)
     alignAsTable = attr.ib(default=False, validator=instance_of(bool))
     hideEmpty = attr.ib(default=False, validator=instance_of(bool))
     hideZero = attr.ib(default=False, validator=instance_of(bool))
     rightSide = attr.ib(default=False, validator=instance_of(bool))
     sideWidth = attr.ib(default=None)
 
+    def convert_values_field_value(self):
+        if self.values is None:
+            self.values = (self.avg or self.current or self.max or self.min)
+
     def to_json_data(self):
+        self.convert_values_field_value()
+
```

Comment:
Sorry to be fussy about this, but I'd much rather something that doesn't change state.

e.g.

```python
values = (self.avg or self.current or self.max or self.min) if self.values is None else self.values
```